src/condor_daemon_client/dc_schedd.cpp

```diff
@@ -2104,3 +2104,277 @@ DCSchedd::requestImpersonationTokenAsync(const std::string &identity,
 	}
 	return true;
 }
+
+// TODO: how do we make this global??
+enum
+{
+	Q_NO_SCHEDD_IP_ADDR = 20,
+	Q_SCHEDD_COMMUNICATION_ERROR,
+	Q_INVALID_REQUIREMENTS,
+	Q_INTERNAL_ERROR,
+	Q_REMOTE_ERROR,
+	Q_UNSUPPORTED_OPTION_ERROR
+};
+
+/*static*/ int DCSchedd::makeUsersQueryAd (
+	classad::ClassAd & request_ad,
+	const char * constraint,
+	const char * projection,
+	bool send_server_time /*= false*/,
+	int match_limit /*= -1*/)
+{
+	if (constraint && constraint[0]) {
+		classad::ClassAdParser parser;
+		classad::ExprTree *expr = NULL;
+		parser.ParseExpression(constraint, expr);
+		if (!expr) return Q_PARSE_ERROR;
+
+		request_ad.Insert(ATTR_REQUIREMENTS, expr);
+	}
+
+	if (projection) {
+		request_ad.InsertAttr(ATTR_PROJECTION, projection);
+	}
+
+	if (send_server_time) {
+		request_ad.Assign(ATTR_SEND_SERVER_TIME, true);
+	}
+
+	if (match_limit >= 0) {
+		request_ad.InsertAttr(ATTR_LIMIT_RESULTS, match_limit);
+	}
+	return 0;
+
+}
+
+/*static*/ int DCSchedd::makeUsersQueryAd (
+	classad::ClassAd & request_ad,
+	const char * constraint,
+	classad::References & attrs,
+	int match_limit /* = -1 */)
+{
+	bool send_server_time = false;
+	std::string projlist;
+	const char * projection = nullptr;
+
+	if ( ! attrs.empty()) {
+		for (auto attr : attrs) {
+			if (projlist.empty()) projlist += "\n";
```

Comment:
```suggestion
			if (! projlist.empty()) projlist += "\n";
```
?

Reply:
yeah, that's busted.  and so is the code in the schedd that reads that attr. oops...